build-contract.sh

```diff
@@ -0,0 +1,115 @@
+#!/bin/bash
+
+# Build script to compile/test the smart contract using the SmartPy CLI. 
+# Includes an optional argument to specify a target directory to copy 
+# the generated contract file. Default target dir is the current directory.
+#
+# Usage: ./build-contract.sh [ /optional/path/to/dir ]
+#
+
+CENTOS_RELEASE=/etc/centos-release
+REDHAT_RELEASE=/etc/redhat-release
+FEDORA_RELEASE=/etc/fedora-release
+LSB_RELEASE=/etc/lsb-release
+ORACLE_RELEASE=/etc/oracle-release
+SYSTEM_RELEASE=/etc/system-release
+DEBIAN_VERSION=/etc/debian_version
+
+PURPLE='\033[0;95m'
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+NC='\033[0m'
+
+function fail() {
+  printf "${RED}[!] $1${NC}\n"
+  exit 1
+}
+
+function console() {
+  printf "${GREEN}[+] $1${NC}\n"
+}
+
+function platform() {
+  local  __out=$1
+  if [[ -f "$LSB_RELEASE" ]] && grep -q 'DISTRIB_ID=Ubuntu' $LSB_RELEASE; then
+    FAMILY="debian"
+    eval $__out="ubuntu"
+  elif [[ -f "$DEBIAN_VERSION" ]]; then
+    FAMILY="debian"
+    eval $__out="debian"
+  elif [[ -f "$FEDORA_RELEASE" ]]; then
+    FAMILY="fedora"
+    eval $__out="fedora"
+  elif [[ -f "$CENTOS_RELEASE" ]]; then
+    FAMILY="centos"
+    eval $__out="centos"
+  elif [[ -f "$REDHAT_RELEASE" ]]; then
+    FAMILY="redhat"
+    eval $__out="redhat"
+  else
+    eval $__out=`uname -s | tr '[:upper:]' '[:lower:]'`
+  fi
+}
+
+function distro() {
+  local __out=$2
+  if [[ $1 = "ubuntu" ]]; then
+    eval $__out=`awk -F= '/DISTRIB_CODENAME/ { print $2 }' $LSB_RELEASE`
+  elif [[ $1 = "darwin" ]]; then
+    eval $__out=`sw_vers -productVersion | awk -F '.' '{print $1 "." $2}'`
+  elif [[ $1 = "debian" ]]; then
+    eval $__out="`cat /etc/os-release | grep 'VERSION=' | cut -c 9-`"
+  else
+    eval $__out="unknown_version"
+  fi
+}
+
+function brew_install() {
+    console "Installing $1"
+    if brew list $1 &>/dev/null; then
+        echo "${1} is already installed"
+    else
+        brew install $1 && echo "$1 is installed"
+    fi
+}
+
+TEMP_DIR=$(shell mktemp -d tmp-XXXX)
```

Comment:
I'm not on any of the supported platforms - so maybe this doesn't apply - but I get `shell: command not found`. 

Also, it looks like `mktemp` has a slightly different interface on OSX v Linux: https://unix.stackexchange.com/a/84980

Reply:
oh i see. will fix. thanks for the pointer!